lib/web/apiserver.go

```diff
@@ -1547,20 +1548,25 @@ func (h *Handler) getResetPasswordToken(ctx context.Context, tokenID string) (in
 		return nil, trace.Wrap(err)
 	}
 
-	// RotateUserTokenSecrets rotates secrets for a given tokenID.
-	// It gets called every time a user fetches 2nd-factor secrets during registration attempt.
+	// CreateRegisterChallenge rotates TOTP secrets for a given tokenID.
+	// It is required to get called every time a user fetches 2nd-factor secrets during registration attempt.
 	// This ensures that an attacker that gains the ResetPasswordToken link can not view it,
 	// extract the OTP key from the QR code, then allow the user to signup with
 	// the same OTP token.
-	secrets, err := h.auth.proxyClient.RotateUserTokenSecrets(ctx, tokenID)
+	res, err := h.auth.proxyClient.CreateRegisterChallenge(ctx, &proto.CreateRegisterChallengeRequest{
+		TokenID:    tokenID,
+		DeviceType: proto.DeviceType_DEVICE_TYPE_TOTP,
+	})
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
+	chal := client.MakeRegisterChallenge(res)
+
 	return ui.ResetPasswordToken{
 		TokenID: token.GetName(),
 		User:    token.GetUser(),
-		QRCode:  secrets.GetQRCode(),
+		QRCode:  chal.TOTPChallenge.QRCodeBytes,
```

Comment:
Get the QRCode bytes directly from `res` and skip the conversion above?

Reply:
duh! :woman_facepalming: